framework/src/base/MooseApp.C

```diff
@@ -1951,11 +1952,16 @@ MooseApp::addRelationshipManager(std::shared_ptr<RelationshipManager> relationsh
 {
   // We don't need Geometric-only RelationshipManagers when we run with
   // ReplicatedMesh unless we are splitting the mesh.
-  if (!_action_warehouse.mesh()->isDistributedMesh() && !_split_mesh &&
-      (relationship_manager->isType(Moose::RelationshipManagerType::GEOMETRIC) &&
-       !(relationship_manager->isType(Moose::RelationshipManagerType::ALGEBRAIC) ||
-         relationship_manager->isType(Moose::RelationshipManagerType::COUPLING))))
-    return false;
+  // We prefer to always add geometric RMs. There is no hurt to add RMs for replicated mesh
+  // since MeshBase::delete_remote_elements{} is not implemented in replicated mesh.
+  // The motivation here is that mesh parallel type might not be set yet for distributed mesh
+  // generator, and for the default parallel type "isDistributedMesh()" is false. All geometric
+  // RMs are just ignored for distributed mesh generator
```

Comment:
The geometric RMs were ignored for distributed mesh generator because of the old logic here:

```
if (!_action_warehouse.mesh()->isDistributedMesh() && ....)
  return false
```

`_action_warehouse.mesh()->isDistributedMesh()`  was false because `MooseApp::addRelationshipManager` was called before we informed moose that the future mesh was a distributed mesh. 

We deleted this old logic in this PR so that we have all geometric  RMs added for the distributed mesh generator

Reply:
Yes very good. I am glad we add geometric ghosting for both replicated and distributed now. Uniformity is nice